src/main/java/com/researchspace/dmptool/model/DMPToolDMP.java

```diff
@@ -1,34 +1,37 @@
 package com.researchspace.dmptool.model;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+import com.fasterxml.jackson.annotation.JsonSetter;
 import java.util.Map;
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import java.util.Optional;
+import com.researchspace.rda.model.DMP;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
 
 @Data
 @NoArgsConstructor
-@AllArgsConstructor
-public class DMPToolDMP {
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonTypeName(value = "dmp")
+public class DMPToolDMP extends DMP {
 
-	private Long id;
-	private String title;
-	private String description;
+	@JsonProperty("dmproadmap_links")
+	private Map<String, String> links;
 
-	@SuppressWarnings("unchecked")
-	@JsonProperty("dmp")
-	private void unpackNested(Map<String,Object> dmp) {
-		this.title = (String) dmp.get("title");
-		this.description = (String) dmp.get("description");
-
-		Map<String,String> links = (Map<String,String>) dmp.get("dmproadmap_links");
-		String getUrlString = links.get("get");
+	/*
+	 * This is... unfortunate. Would be much better for us to be using the
+	 * proper dmp_id property as a unique identifier, rather than the
+	 * internal implementation detail used by DMPTool. However, it is now
+	 * used as a database key (UserDMP) so removing this without breaking
+	 * backwards compatibility will be difficult.
+	 */
```

Comment:
we can make database updates to move things around, if really needed.

Reply:
Ok, re-worded comment. No need to do that now, but might be worth it in the future if we do any more work on the integration